### PR TITLE
Add AtCoder contest format

### PR DESCRIPTION
**Pull request by kiritofeng**

### CODE REVIEW

**Comment by kiritofeng** [comment]
Question: How should we represent the time penalty? A small `-1`, (or however many submissions) beside the score, in red?

File: judge/contest_format/atcoder.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                    WHERE ccs.problem_id = cp.id AND ccs.participation_id = %s AND ccs.points = MAX(cs.points)
+            ) AS `time`, cp.id AS `prob`
+            FROM judge_contestproblem cp INNER JOIN
+                 judge_contestsubmission cs ON (cs.problem_id = cp.id AND cs.participation_id = %s) LEFT OUTER JOIN
+                 judge_submission sub ON (sub.id = cs.submission_id)
+            GROUP BY cp.id
+            ''', (participation.id, participation.id))
+
+            for score, time, prob in cursor.fetchall():
+                    time = from_database_time(time)
```
Comment (xiaowuc1) [inline judge/contest_format/atcoder.py]: nit: indentation is off

File: judge/contest_format/atcoder.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    dt = (time - participation.start).total_seconds()
+
+                    if score:
+                        # Compute penalty
+                        if self.config['penalty']:
+                            prev = participation.submissions.exclude(submission__result__in=['IE', 'CE']) \
+                                                            .filter(problem_id=prob, submission__date__lte=time).count() - 1
+                            penalty += prev * self.config['penalty'] * 60
+                        else:
+                            prev = 0
```
Comment (xiaowuc1) [inline judge/contest_format/atcoder.py]: if `prev` is the number of incorrect submissions and used as an indicator of penalty, this should not be gated by score being present - instead the number of incorrect submissions should be included and noted with a score of zero (see page 16 of https://atcoder.jp/contests/abc131/standings at ranks 301 and 303 for an example in their production system)

File: judge/contest_format/atcoder.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+                format_data[str(prob)] = {'time': dt, 'points': score, 'penalty': prev}
+                points += score
+
+        participation.cumtime = cumtime + penalty
+        participation.score = points
+        participation.format_data = format_data
+        participation.save()
+
+def display_user_problem(self, participation, contest_problem):
```
Comment (xiaowuc1) [inline judge/contest_format/atcoder.py]: indentation is off

File: judge/contest_format/atcoder.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        format_data = (participation.format_data or {}).get(str(contest_problem.id))
+        if format_data:
+            penalty = format_html('<small style="color:red"> -{penalty}</small>',
+                                penalty=floatformat(format_data['penalty'])) if format_data['penalty'] else ''
```
Comment (xiaowuc1) [inline judge/contest_format/atcoder.py]: indentation is off

also is that going to render as `-number of wrong submissions` or `-(number of wrong submissions)`? Ideally do the latter.

**Comment by kiritofeng** [comment]
Preview
Preview

**Pull request merged**